quilt/tools/build.py

```diff
@@ -96,22 +100,23 @@ def _file_to_spark_data_frame(ext, path, target, user_kwargs):
     return df
 
 def _file_to_data_frame(ext, path, target, user_kwargs):
-    ext = ext.lower() # ensure that case doesn't matter
-    platform = TARGET.get(target)
-    if platform is None:
-        raise BuildException('Unsupported target platform: %s' % target)
-    logic = platform.get(ext)
-    if logic is None:
-        raise BuildException(
-            "Unsupported transform: %s. Try setting a 'transform' key." % ext)
-    fname = logic['attr']
+    logic = PARSERS.get(ext)
+    module_name = logic['module']
+    the_module = None
+    if module_name in sys.modules:
+        the_module = sys.modules[module_name]
+    else:
+        the_module = importlib.import_module(module_name) 
```

Comment:
I don't really like the idea of importing arbitrary modules: e.g., you might run arbitrary code just by building a package. Maybe whitelist module names?

Reply:
OK but this is in our code in const.py and if they can modify PARSERS they can modify the whitelist. I can see your objection if the modules are specified in build.yml but that's not the case right?

Reply:
Oh whoops, I misunderstood - thought it came from build.yml. That's fine then.